scan/src/components/oracle-script/OracleScriptBridgeCode.re

```diff
@@ -0,0 +1,213 @@
+        src={
+          switch (icon) {
+          | "Ethereum" => Images.ethereumIcon
+          | "Cosmos IBC" => Images.cosmosIBCIcon
+          | "Kadena" => Images.kadenaIcon
+          | _ => Images.missingIcon
+          }
+        }
+      />
+    </div>;
+  };
+};
+
+module LanguageIcon = {
+  [@react.component]
+  let make = (~icon) => {
+    <div className=Styles.iconWrapper>
+      <img
+        className=Styles.iconBody
+        src={
+          switch (icon) {
+          | "Solidity" => Images.solidityIcon
+          | "Vyper" => Images.vyperIcon
+          | "Go" => Images.golangIcon
+          | "PACT" => Images.pactIcon
+          | _ => Images.missingIcon
+          }
+        }
+      />
+    </div>;
+  };
+};
+
+[@react.component]
+let make = () => {
+  let description = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent aliquet tempor imperdiet. Morbi tincidunt molestie tortor a finibus. Nulla hendrerit iaculis metus, in laoreet tellus eleifend vel. Aliquam pretium porta mi, a efficitur justo ullamcorper sed. Donec interdum accumsan nibh, sed tempor lectus rutrum ac. Morbi et magna in magna varius iaculis. Praesent mollis nulla non arcu ullamcorper, at bibendum nibh pellentesque. Aenean ac quam eget turpis euismod lacinia. Phasellus libero lectus, pulvinar at ipsum non, ullamcorper commodo felis.";
+  let codetest = {j|
+    pragma solidity ^0.5.0;
+
+    import "./Borsch.sol";
+
+    library ResultDecoder {
+      using Borsh for Borsh.Data;
+
+      struct Result {
+        string symbol;
+        uint64 multiplier;
+        uint8 what;
+      }
+
+      function decodeResult(bytes memory _data)
+        internal
+        pure
+        returns (Result memory result)
+      {
+          Borsh.Data memory data = Borsh.from(_data);
+          result.symbol = string(data.decodeBytes());
+          result.multiplier = data.decodeU64();
+          result.what = data.decodeU8();
+      }
+    }|j};
+
+  let (targetPlatform, setTargetPlatform) = React.useState(_ => "Ethereum");
+  let (language, setLanguage) = React.useState(_ => "Solidity");
+  <div className=Styles.tableWrapper>
+    <>
+      <VSpacing size={`px(10)} />
+      <Row>
+        <HSpacing size={`px(15)} />
+        <Col>
+          <div> <Text value="Target Platform" /> </div>
+          <VSpacing size={`px(5)} />
+          <div className=Styles.selectWrapper>
+            <TargetPlatformIcon icon=targetPlatform />
+            <select
+              className=Styles.selectContent
+              onChange={event => {
+                let newValue = ReactEvent.Form.target(event)##value;
+                setTargetPlatform(_ => newValue);
+                switch (newValue) {
+                | "Ethereum" => setLanguage(_ => "Solidity")
+                | "Cosmos IBC" => setLanguage(_ => "Go")
+                | "Kadena" => setLanguage(_ => "PACT")
+                | _ => setLanguage(_ => "Solidity")
+                };
```

Comment:
IMO, we should use the variant instead of the string. Let's make a type for target platform and language.